packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -417,7 +417,7 @@ class AppDomain extends Domain {
         );
         _sendAppEvent(app, 'stop');
       } catch (error) {
-        _sendAppEvent(app, 'stop', <String, dynamic>{'error': error.toString()});
+        _sendAppEvent(app, 'stop', <String, dynamic>{'error': _toJsonable(error)});
```

Comment:
If the plugin will only ever show these errors in the console, that works fine.  If and when we ever want to add more semantic error handling in the plugin, we may want to revisit our error encoding.